loader/extends.go

```diff
@@ -71,14 +71,15 @@ func ApplyExtends(ctx context.Context, dict map[string]any, workingdir string, o
 				if err != nil {
 					return err
 				}
-				relworkingdir := filepath.Dir(local)
-				if !filepath.IsAbs(local) {
-					relworkingdir, err = filepath.Rel(workingdir, relworkingdir)
-					if err != nil {
-						return err
-					}
-				}
+				localdir := filepath.Dir(local)
+				relworkingdir := loader.Dir(path)
+
 				extendsOpts := opts.clone()
+				extendsOpts.ResourceLoaders = append([]ResourceLoader{}, opts.ResourceLoaders...)
+				// replace localResourceLoader with a new flavour, using extended file base path
+				extendsOpts.ResourceLoaders[len(opts.ResourceLoaders)-1] = localResourceLoader{
+					WorkingDir: localdir,
+				}
 				extendsOpts.ResolvePaths = true
```

Comment:
This line is the root cause of the https://github.com/docker/compose/issues/11377
It was introduced to remove potential import cycles https://github.com/compose-spec/compose-go/commit/714c186dd139b0e732a9441c4445a6836a8d10d4

Reply:
`ResolvePaths` MUST be set so that `./env` in a compose file that is used by `extends: file=./foo/bar/compose.yaml` is resolved as `./foo/bar/env` and can be used as expected by the caller

Used by extend, PathResolution resolves _relative_ paths from working directory, which is itself relative